// react-alerts-plus/lib/AlertContainer.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var AlertContainer = function AlertContainer(_ref) {
  var children = _ref.children,
      style = _ref.style;
  return _react["default"].createElement("div", {
    style: style,
    className: "alertContainer",
    "data-testid": "alert-container-test"
  }, children);
};

AlertContainer.propTypes = {
  children: _propTypes["default"].oneOfType([_propTypes["default"].arrayOf(_propTypes["default"].node), _propTypes["default"].node]).isRequired,
  style: _propTypes["default"].shape({
    position: _propTypes["default"].string.isRequired,
    top: _propTypes["default"].oneOfType([_propTypes["default"].string, _propTypes["default"].number]),
    right: _propTypes["default"].oneOfType([_propTypes["default"].string, _propTypes["default"].number]),
    bottom: _propTypes["default"].oneOfType([_propTypes["default"].string, _propTypes["default"].number]),
    left: _propTypes["default"].oneOfType([_propTypes["default"].string, _propTypes["default"].number]),
    transform: _propTypes["default"].string,
    zIndex: _propTypes["default"].number
  }).isRequired
};
var _default = AlertContainer;
exports["default"] = _default;
