// react-alerts-plus/lib/getPosition.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var getPosition = function getPosition(position, offset) {
  var POSITION = 'fixed';
  var ZINDEX = 100;
  var TRANSFORM = 'translate(-50%, 0%)';
  var CENTER_PERCENT = "50%";

  switch (position) {
    case 'top left':
      return {
        position: POSITION,
        top: "".concat(offset),
        left: "".concat(offset),
        zIndex: ZINDEX
      };

    case 'top right':
      return {
        position: POSITION,
        top: "".concat(offset),
        right: "".concat(offset),
        zIndex: ZINDEX
      };

    case 'bottom left':
      return {
        position: POSITION,
        bottom: "".concat(offset),
        left: "".concat(offset),
        zIndex: ZINDEX
      };

    case 'bottom right':
      return {
        position: POSITION,
        bottom: "".concat(offset),
        right: "".concat(offset),
        zIndex: ZINDEX
      };

    case 'top center':
      return {
        position: POSITION,
        top: "".concat(offset),
        left: CENTER_PERCENT,
        transform: TRANSFORM,
        zIndex: ZINDEX
      };

    case 'bottom center':
      return {
        position: POSITION,
        bottom: "".concat(offset),
        left: CENTER_PERCENT,
        transform: TRANSFORM,
        zIndex: ZINDEX
      };

    default:
      return null;
  }
};

var _default = getPosition;
exports.default = _default;
