// react-alerts-plus/lib/getPositionStyles.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var getPositionStyles = function getPositionStyles(position, offset, isMobile) {
  var zIndex = arguments.length > 3 && arguments[3] !== undefined ? arguments[3] : 100;
  var POSITION = 'fixed';
  var TRANSFORM = 'translate(-50%, 0%)';
  var CENTER_PERCENT = "50%";

  switch (position) {
    case 'topLeft':
      if (!isMobile) {
        return {
          position: POSITION,
          top: "".concat(offset),
          left: "".concat(offset),
          zIndex: zIndex
        };
      }

      return {
        position: POSITION,
        top: '0px',
        right: '0px',
        left: '0px',
        zIndex: zIndex
      };

    case 'topRight':
      if (!isMobile) {
        return {
          position: POSITION,
          top: "".concat(offset),
          right: "".concat(offset),
          zIndex: zIndex
        };
      }

      return {
        position: POSITION,
        top: '0px',
        right: '0px',
        left: '0px',
        zIndex: zIndex
      };

    case 'bottomLeft':
      if (!isMobile) {
        return {
          position: POSITION,
          bottom: "".concat(offset),
          left: "".concat(offset),
          zIndex: zIndex
        };
      }

      return {
        position: POSITION,
        bottom: '0px',
        right: '0px',
        left: '0px',
        zIndex: zIndex
      };

    case 'bottomRight':
      if (!isMobile) {
        return {
          position: POSITION,
          bottom: "".concat(offset),
          right: "".concat(offset),
          zIndex: zIndex
        };
      }

      return {
        position: POSITION,
        bottom: '0px',
        right: '0px',
        left: '0px',
        zIndex: zIndex
      };

    case 'topCenter':
      if (!isMobile) {
        return {
          position: POSITION,
          top: "".concat(offset),
          left: CENTER_PERCENT,
          transform: TRANSFORM,
          zIndex: zIndex
        };
      }

      return {
        position: POSITION,
        top: '0px',
        right: '0px',
        left: '0px',
        zIndex: zIndex
      };

    case 'bottomCenter':
      if (!isMobile) {
        return {
          position: POSITION,
          bottom: "".concat(offset),
          left: CENTER_PERCENT,
          transform: TRANSFORM,
          zIndex: zIndex
        };
      }

      return {
        position: POSITION,
        bottom: '0px',
        right: '0px',
        left: '0px',
        zIndex: zIndex
      };

    default:
      return null;
  }
};

var _default = getPositionStyles;
exports["default"] = _default;
